File: sort.py
import numpy as np
import time

soday = 10
sophantu = 1000000


with open("data.txt", "r") as f:
    data = np.fromfile(f, sep=" ")


if len(data) != soday * sophantu:
    raise ValueError("Số phần tử trong data.txt không đúng!")

data = data.reshape((soday, sophantu))  

sorted_arrays = []
for i in range(soday):
    start = time.time()
    
    sorted_arr = np.sort(data[i])
    
    end = time.time()
    duration = (end - start) * 1000  
    
    sorted_arrays.append(sorted_arr)
    
    print(f"Lần {i + 1}: {duration:.2f} ms", file=sys.stderr)


np.savetxt("pysorted.txt", sorted_arrays, fmt="%.6f")

File: sort.cpp
#include <iostream>
#include <cstdio>
#include <vector>
#include <algorithm>
#include <chrono> 

using namespace std;
using namespace chrono; 

const int soday = 10;
const int sophantu = 1000000;


int main() {
    freopen("data.txt", "r", stdin);
    freopen("csorted.txt", "w", stdout);

    vector<double> a(sophantu);

    for (int i = 0; i < soday; i++) {
        for (int j = 0; j < sophantu; j++) {
            if (!(cin >> a[j])) {
                cerr << "Lỗi đọc dữ liệu!" << endl;
                return 1;
            }
        }

       
        auto start = high_resolution_clock::now();
        
        sort(a.begin(), a.end());

        
        auto end = high_resolution_clock::now();
        double duration = duration_cast<milliseconds>(end - start).count();

        
        for (int j = 0; j < sophantu; j++) {
            cout << a[j] << " ";
        }
        cout << endl;

        
        cerr << "Lần " << i + 1 << ": " << duration << " ms" << endl;
    }

    return 0;
}
